crates/ruff/src/settings/line_width.rs

```diff
@@ -0,0 +1,205 @@
+use ruff_cache::CacheKey;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use unicode_width::UnicodeWidthChar;
+
+use ruff_macros::CacheKey;
+
+pub trait LineWidthState: Clone + Default + CacheKey {}
+
+#[derive(Clone, Copy, Debug, CacheKey)]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+pub struct Length;
+
+impl Default for Length {
+    fn default() -> Self {
+        Self
+    }
+}
+
+#[derive(Clone, Copy, Default, CacheKey)]
+pub struct TabInfos {
+    column: usize,
+    tab_size: TabSize,
+}
+
+impl LineWidthState for Length {}
+impl LineWidthState for TabInfos {}
+
+#[derive(Clone, Copy, Debug, CacheKey)]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+pub struct LineWidth<S: LineWidthState> {
+    width: usize,
+    extra: S,
+}
+
+impl Default for LineWidth<Length> {
+    fn default() -> Self {
+        Self {
+            width: 88,
```

Comment:
What's the reason that we initialize the width to `88`? Or what's the meaning of the `width` field?

Reply:
`LineWidth<Length>` or `LineLength` is the type representing line length settings.
`88` comes from `LINE_LENGTH`.